### ISSUE TITLE
Error when loading tools without internet connection

### ISSUE THREAD

**Issue by fwinkelbauer** [cake-build/cake#2113]
### What You Are Seeing?

Cake fails if it cannot get a connection to a nuget source when installing a nuget package (even if the package is already present).

### What is Expected?

Cake should work without an internet connection if the appropiate package is already present.

### What version of Cake are you using?

0.26.1

### Are you running on a 32 or 64 bit system?

64 bit

### What environment are you running on?  Windows? Linux? Mac?

Windows 7

### Are you running on a CI Server?  If so, which one?

Local

### How Did You Get This To Happen? (Steps to Reproduce)

[Cake script](https://github.com/fwinkelbauer/Bumpy/blob/master/Source/build.cake):

- Run `cake` or `build.ps1`
- Disconnect from the internet
- Run `cake` or `build.ps1` again

### Output Log

Error messages include: `An error occurred while retrieving package metadata for 'Cake.Bumpy' from source 'nuget.org'.` or ` An error occurred while retrieving package metadata for 'octokit' from source 'nuget.org'`

```
Installing tools...
  GET https://api.nuget.org/v3/registration3-gz-semver2/vswhere/index.json
An error was encountered when fetching 'GET https://api.nuget.org/v3/registration3-gz-semver2/vswhere/index.json'. The request will now be retried.
Fehler beim Senden der Anforderung.
  Der Remotename konnte nicht aufgelöst werden: 'api.nuget.org'
  GET https://api.nuget.org/v3/registration3-gz-semver2/vswhere/index.json
An error was encountered when fetching 'GET https://api.nuget.org/v3/registration3-gz-semver2/vswhere/index.json'. The request will now be retried.
Fehler beim Senden der Anforderung.
  Der Remotename konnte nicht aufgelöst werden: 'api.nuget.org'
  GET https://api.nuget.org/v3/registration3-gz-semver2/vswhere/index.json
Error: System.AggregateException: Mindestens ein Fehler ist aufgetreten. ---> NuGet.Protocol.Core.Types.FatalProtocolException: An error occurred while retrieving package metadata for 'vswhere' from source 'nuget.org'. ---> System.Net.Http.HttpRequestException: Fehler beim Senden der Anforderung. ---> System.Net.WebException: Der Remotename konnte nicht aufgelöst werden: 'api.nuget.org'
```

**Comment by mholo65** [comment]
@fwinkelbauer you don't pin the version of `Cake.Bumpy` addin.

[Current behaviour](https://github.com/cake-build/cake/blob/447f3722fa779528940a91af84912e5ba4298cde/src/Cake.NuGet/Install/NuGetPackageInstaller.cs#L184) is to query the package sources to get the most recent version. This way the latest version is always installed.

**Comment by jnm2** [comment]
Huh. So if I had known to pin all tool versions, I would have been able to use Cake on my last flights?

**Comment by mholo65** [comment]
@jnm2 in latest beta bits it ***should*** be possible, if you are not relying on loading dependencies.... Currently trying to sort out some edge cases... Hopefully I will get this done by 0.27.0

**Comment by jnm2** [comment]
Thanks! Good to know. Got most of my flying out of the way now, so don't worry on my account!
It would make sense to me to issue a warning rather than failing if none of the package sources can be contacted but all the packages can be satisfied from your local cache.

**Comment by fwinkelbauer** [comment]
Well TIL, thanks!

Just to make sure that I got your point: The current behavior will change in 0.27.0 and so I should leave this issue open, right?

**Comment by mholo65** [comment]
@fwinkelbauer yes, leave this issue open. I'll change the behavior to "log warning and continue" if we for some reason are unable to communicate with a remote source. I'm also investigating, why we are unable to read dependency info from a local package source (e.g. `tools/Addins` or `~/.nuget/packages`)...

Hopefully I'll get this all fixed in 0.27.0.
